DOCS-5144 Adds Build an Integration, Build a Marketplace Offering, and Create a Tile Documentation (#14297)
DataDog/integrations-core: docs/dev/create_a_tile.md

  text: "Learn how to build a Marketplace offering"
- link: "/developers/integrations/oauth_for_integrations"
  tag: "Documentation"
  text: "Learn about using OAuth for integrations"
---

## Overview

This page walks you through how to develop an offering on the Datadog Marketplace. If you have any questions, reach out to <a href="mailto:marketplace@datadoghq.com">marketplace@datadoghq.com</a>.

{{% tab "Paid integrations" %}}
## Set up a directory and clone the Marketplace repository

Once you've decided on an offering, set up a directory:

1. Request access to the [Marketplace repository][18] by following the instructions in the [Marketplace documentation][19].
2. Create a `dd` directory:
   {{< code-block lang="shell" >}}mkdir $HOME/dd{{< /code-block >}}

   The Datadog Development Toolkit command expects you to be working in the `$HOME/dd/` directory. This is not mandatory, but working in a different directory requires additional configuration steps.
3. Once you have been granted access to the Marketplace repository, create the `dd` directory and clone the `marketplace` repo:
   {{< code-block lang="shell" >}}git clone git@github.com:DataDog/marketplace.git{{< /code-block >}}

> 4. Create a feature branch to work in.

Reviewer: It may just be my screen, but it looks like number 4 is a different font size the 1, 2, 3, etc.

| Description        | The maximum number of characters allowed is 300.                                                                                                  |

Follow this template to define the `media` object in the `manifest.json` file which includes an image, a video thumbnail, and a video:

{{< code-block lang="json" filename="manifest.json" collapsible="true" >}}
"media": [
      {
        "media_type": "image",
        "caption": "A Datadog Marketplace Integration OOTB Dashboard",
        "image_url": "images/integration_name_image_name.png"
      },
      {
        "media_type": "video",
        "caption": "A Datadog Marketplace Integration Overview Video",
        "image_url": "images/integration_name_video_thumbnail.png",
        "vimeo_id": 123456789
      },
    ],
{{< /code-block >}}

For more information, see [Integrations Assets Reference][22].

Before you open a pull request, run the following command to catch any problems with your integration:
```
ddev validate all <INTEGRATION_NAME>
```

## Open a pull request

> Push up your feature branch and open a pull request that contains your integration tile's asset files (including images) in the [`marketplace`][18] or [`integrations-extras`][number] repository. After you've created your pull request, automatic checks will run to verify that your pull request is in good shape and contains all the required content to be updated.

Author: ```suggestion
Push up your feature branch and open a pull request that contains your integration tile's asset files (including images) in the [`marketplace`][18] or [`integrations-extras`][26] repository. After you've created your pull request, automatic checks will run to verify that your pull request is in good shape and contains all the required content to be updated.
```

[7]: https://docs.datadoghq.com/api/latest/events/
[8]: https://docs.datadoghq.com/api/latest/service-checks/
[9]: https://docs.datadoghq.com/tracing/guide/send_traces_to_agent_by_api/
[10]: https://docs.datadoghq.com/api/latest/incidents/
[11]: https://docs.datadoghq.com/api/latest/security-monitoring/
[12]: https://docs.datadoghq.com/developers/integrations/
[13]: https://docs.datadoghq.com/developers/#creating-your-own-solution
[14]: https://docs.datadoghq.com/api/latest/
[15]: https://docs.datadoghq.com/developers/integrations/oauth_for_integrations/
[16]: https://docs.datadoghq.com/developers/datadog_apps/
[17]: https://app.datadoghq.com/apps/
[18]: https://github.com/Datadog/marketplace
[19]: https://docs.datadoghq.com/developers/integrations/marketplace_offering/#request-access-to-marketplace
[20]: https://www.python.org/downloads/
[21]: https://pypi.org/project/datadog-checks-dev/
[22]: https://docs.datadoghq.com/developers/integrations/check_references/#manifest-file
[23]: https://datadoghq.com/blog/
[24]: https://github.com/DataDog/integrations-extras/tree/master/vantage
[25]: https://docs.datadoghq.com/developers/integrations/python

> [26]: Integrations page

Author: ```suggestion
[26]: https://github.com/Datadog/integrations-extras
```

---
title: Create a Tile
type: documentation
description: Learn how to develop and publish an integration tile.
aliases:
- 'developers/marketplace/offering'
dependencies: "https://github.com/DataDog/integrations-core/blob/alai97/add-marketplace-documentation/docs/dev/create_a_tile.md"
further_reading:
- link: "https://partners.datadoghq.com/"
  tag: "Partner Network"
  text: "Datadog Partner Network"
- link: "https://www.datadoghq.com/blog/datadog-marketplace/"
  tag: "Blog"
  text: "Expand your monitoring reach with the Datadog Marketplace"
- link: "/developers/integrations/marketplace_offering/"
  tag: "Documentation"
  text: "Learn how to build a Marketplace offering"
- link: "/developers/integrations/oauth_for_integrations"
  tag: "Documentation"
  text: "Learn about using OAuth for integrations"
---

## Overview

> This page walks you through how to develop an offering on the Datadog Marketplace. If you have any questions, reach out to <a href="mailto:marketplace@datadoghq.com">marketplace@datadoghq.com</a>.

Reviewer 2: @anawishnoff Lmk what you think: 

 This page walks you through how to develop an offering on the Datadog Marketplace. A tile-only listing can be listed as a paid offering within Marketplace or as an out-of-the-box integration within integrations-extras. 

A tile-only listing is for any integration that does not use the Datadog Agent and is informational only. This type of listing does not include code as the code is hosted on the partners end.

> This page walks you through how to develop an offering on the Datadog Marketplace. If you have any questions, reach out to <a href="mailto:marketplace@datadoghq.com">marketplace@datadoghq.com</a>.

Reviewer 3: This page is for both free and paid offerings, so i'd change it to:

This page walks you through how to develop an offering that will be displayed in a tile on the Datadog Integrations page or on the Datadog Marketplace. For all offerings that do not use the Datadog Agent, including API-based integrations, professional services listings, software licenses, and Datadog Apps, a **tile-only listing** is required. This means that there is no Agent code attached to the tile, but it serves as an informative point of entry where customers can learn more about your offering, learn how to set it up, as well as install or purchase your offering to unlock  out-of-the-box dashboards and other assets.

> {{% tab "Marketplace Offerings" %}}

Reviewer 2: This includes the types of offerings

> {{% tab "Marketplace Offerings" %}}

Reviewer 2: It has :  https://a.cl.ly/eDure0er

After you install the Developer tool, configure it for the `integrations-extras` repo:

Set `integrations-extras` as the default working repository:

{{< code-block lang="shell" >}}
ddev config set extras $HOME/dd/integrations-extras
ddev config set repo extras
{{< /code-block >}}

If you used a directory other than `$HOME/dd` to clone the integrations-extras directory, use the following command to set your working repository:

{{< code-block lang="shell" >}}
ddev config set extras <PATH/TO/INTEGRATIONS_EXTRAS>
ddev config set repo extras
{{< /code-block >}}

## Populate the integration tile scaffolding

Run the `ddev` command to create scaffolding for an informational tile-only listing:

1. Make sure you're inside the `integrations-extras` directory:
   {{< code-block lang="shell" >}}cd $HOME/dd/integrations-extras{{< /code-block >}}
2. Run the `ddev` command with the `-t tile` option
   {{< code-block lang="shell" >}}ddev create -t tile "<Offering Name>"{{< /code-block >}}
   

> The options you use with the `ddev` command are different depending on what type of integration you are developing. For a full list of the files created by the `ddev` command, see [Integrations assets][20].   

Reviewer 2: @alai97 I believe all commands are now the same for integrations and marketplace based off the work @hestonhoffman and I did last quarter. Correct?

> The options you use with the `ddev` command are different depending on what type of integration you are developing. For a full list of the files created by the `ddev` command, see [Integrations assets][20].   

Reviewer 4: It's technically still true, but I don't think the sentence adds a lot of value since we made those changes, so I think it's okay to remove it.

> The options you use with the `ddev` command are different depending on what type of integration you are developing. For a full list of the files created by the `ddev` command, see [Integrations assets][20].   

Reviewer 2: Thanks! removing

| Description        | The maximum number of characters allowed is 300.                                                                                                  |

Follow this template to define the `media` object in the media carousel which includes an image, a video thumbnail, and a video:

{{< code-block lang="json" filename="manifest.json" collapsible="true" >}}
"media": [
      {
        "media_type": "image",
        "caption": "A Datadog Marketplace Integration OOTB Dashboard",
        "image_url": "images/integration_name_image_name.png"
      },
      {
        "media_type": "video",
        "caption": "A Datadog Marketplace Integration Overview Video",
        "image_url": "images/integration_name_video_thumbnail.png",
        "vimeo_id": 123456789
      },
    ],
{{< /code-block >}}

For more information, see [Integrations Assets Reference][22].

Before you open a pull request, run the following command to catch any problems with your integration:
```
ddev validate all <INTEGRATION_NAME>
```

## Open a pull request

> Push up your feature branch and open a pull request that contains your integration tile's asset files (including images) in the [`marketplace` repository][18]. The Marketplace repository does not allow forks. For instructions on creating a clone of the repo, see the [Set up section](#set-up-a-directory-and-clone-the-marketplace-repository). After you've created your pull request, automatic checks run in Azure DevOps pipelines to verify that your pull request is in good shape and contains all the required content to be updated.

Reviewer 2: @alai97 Is there a reason why we specify Azure DevOps pipeline here?

ddev config set repo extras
{{< /code-block >}}

If you used a directory other than `$HOME/dd` to clone the integrations-extras directory, use the following command to set your working repository:

{{< code-block lang="shell" >}}
ddev config set extras <PATH/TO/INTEGRATIONS_EXTRAS>
ddev config set repo extras
{{< /code-block >}}

## Populate the integration tile scaffolding

For Datadog API integrations and custom dashboard widgets that will be available out-of-the-box on the [Integrations page][102], use the Datadog Development Toolkit to create scaffolding for an informational tile-only listing.

1. Make sure you're inside the `integrations-extras` directory:
   {{< code-block lang="shell" >}}cd $HOME/dd/integrations-extras{{< /code-block >}}
2. Run the `ddev` command with the `-t tile` option
   {{< code-block lang="shell" >}}ddev create -t tile "<Offering Name>"{{< /code-block >}}

[101]: https://docs.datadoghq.com/developers/integrations/python
[102]: https://github.com/Datadog/integrations-extras

{{% /tab %}}
{{< /tabs >}}

## Complete the necessary integration asset files

Make sure that the following required assets for your integration are complete:

> {{% integration-assets %}}

Reviewer 5: Not sure how to leave a comment on this table but this sounds odd
<img width="816" alt="Screen Shot 2023-04-25 at 8 44 44 AM" src="https://user-images.githubusercontent.com/109224841/234331443-4afa7a41-e624-44c8-8fca-54fe4a24c830.png">

Reviewer 2: @alai97 can you help with updating this?

Author: Ahh yes! So sorry I missed this comment. @eho1307 would you want to update it to "definition" instead? Sorry, I default to front matter because of Hugo (https://gohugo.io/content-management/front-matter/).

Reviewer 3: Erica's OOO, but yes let's change it to definition! Thank you

| Setup | Include all the steps required to set up your offering that includes information divided into H3 headings (`###`). Standard topics include:<br><br>- Installing the integration using the in-app integration tile. <br>- Configuring the integration with the appropriate roles and permissions in your Datadog organization.<br>- Accessing out-of-the-box Datadog features that users who purchased and installed the integration can access (such as metrics, events, monitors, logs, dashboards, and more).|
| Uninstallation | Include all the steps for uninstalling your offering. This information is displayed in the **Configure** tab on the tile.|
| Data Collected  | Specify the types of data collected by your integration (if applicable), including events, service checks, logs, etc. Metrics added to the `metadata.csv` file will automatically appear in this tab.  <br><br> If your offering does not provide any of this data, you do not need to add a Data Collected section. |

> | Support | Provide contact information that includes an email to your Support team, a phone number to your company, a link to your company's documentation or blog post, and more help information in a bulleted list format. |

Reviewer 5: Do we actually want to recommend they include a phone number?

Reviewer 2: removed

| Description        | The maximum number of characters allowed is 300.                                                                                                  |

Follow this template to define the `media` object in the `manifest.json` file which includes an image, a video thumbnail, and a video:

{{< code-block lang="json" filename="manifest.json" collapsible="true" >}}
"media": [
      {
        "media_type": "image",
        "caption": "A Datadog Marketplace Integration OOTB Dashboard",
        "image_url": "images/integration_name_image_name.png"
      },
      {
        "media_type": "video",
        "caption": "A Datadog Marketplace Integration Overview Video",
        "image_url": "images/integration_name_video_thumbnail.png",
        "vimeo_id": 123456789
      },
    ],
{{< /code-block >}}

For more information, see [Integrations Assets Reference][22].

Before you open a pull request, run the following command to catch any problems with your integration:
```
ddev validate all <INTEGRATION_NAME>
```

## Open a pull request

> Push up your feature branch and open a pull request that contains your integration tile's asset files (including images) in the [`marketplace`][18] or [`integrations-extras`][26] repository. After you've created your pull request, automatic checks will run to verify that your pull request is in good shape and contains all the required content to be updated.

Reviewer 5: What does 'push up your feature branch' mean?

Reviewer 2: In github, after you commit your changes on your feature branch, you need to push the changes to Datadog

## Review process

Once your pull request passes all checks, reviewers from the `Datadog/agent-integrations`, `Datadog/marketplace-review`, and `Datadog/documentation` teams provide suggestions and feedback on best practices.

> Once you have addressed the feedback and re-requested reviews, these reviewers approve your pull request. Contact the Marketplace team if you would like to preview the tile in your sandbox account. This allows you to validate and preview additional changes on the tile before your pull request is merged.

Reviewer 5: ```suggestion
Once you have addressed the feedback and re-requested reviews, these reviewers will approve your pull request. Contact the Marketplace team if you would like to preview your offering in your sandbox account. This allows you to validate and preview your offering before your pull request is merged.
```
Just a suggestion

Reviewer 2: My concern is since this is focused on tile that it will confused partners by changing it to offering. @anawishnoff Thoughts on this? I changed the rest of the wording to match

Reviewer 2: Keeping tile instead of offering

### How to resolve common validation errors 

Out-of-the-box integrations in the `integrations-extras` repository can run into validation errors when the forked repository is out of date with the origin. Follow the steps below to resolve the validation errors by rebasing. 

Updating the forked repository via the Web App

1. Go to github.com
2. Go to your repositories
3. Select your forked repo of integrations-extras
4. Go to "sync fork" in the top right corner 
5. Click "update branch" 

To rebase and push changes:

1. `git checkout <your working branch>`

2. `git rebase master`

-  If there are any merge conflicts, you'd resolve them here

3. `git push origin <working branch> -f`

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

> 

Reviewer 5: Just curious if this is this the end. Are we recommending them to read anything else like the GTM support for Marketplace?

Reviewer 2: @anawishnoff I think we should include GTM support in the build an offering then link there from here

Reviewer 2: (header 3) GTM opportunities

If you would like to publish this on the Datadog Marketplace, check out our Go-To-Market opportunities here. <link to gtm section>

- link: "/developers/integrations/marketplace_offering/"
  tag: "Documentation"
  text: "Learn how to build a Marketplace offering"
- link: "/developers/integrations/oauth_for_integrations"
  tag: "Documentation"
  text: "Learn about using OAuth for integrations"
---

## Overview

This page walks you through how to develop an offering that will be displayed in a tile on the Datadog Integrations page or on the Datadog Marketplace. For all offerings that do not use the Datadog Agent, including API-based integrations, professional services listings, and software licenses, a tile-only listing is required. This means that there is no Agent code attached to the tile, but it serves as an informative point of entry where customers can learn more about your offering, learn how to set it up, as well as install or purchase your offering to unlock out-of-the-box dashboards and other assets. You can see an example of a tile below:

{{< img src="developers/marketplace/marketplace-tile-example.png" alt="Example Marketplace tile" style="width:30%" >}}

**Select an option below to get started and create a tile on either the Marketplace or Integrations page:** 

{{< tabs >}}

> {{% tab "Build a tile on the Marketplace" %}}

Reviewer 5: Should the integrations option be the default?

## Set up a directory and clone the Marketplace repository

Set up a directory:

1. Request access to the [Marketplace repository][101] by following the instructions in the [Marketplace documentation][102].
2. Create a `dd` directory:
   {{< code-block lang="shell" >}}mkdir $HOME/dd{{< /code-block >}}

   The Datadog Development Toolkit command expects you to be working in the `$HOME/dd/` directory. This is not mandatory, but working in a different directory requires additional configuration steps.
3. Once you have been granted access to the Marketplace repository, create the `dd` directory and clone the `marketplace` repo:
   {{< code-block lang="shell" >}}git clone git@github.com:DataDog/marketplace.git{{< /code-block >}}

> 4. Create a feature branch to work in.

Reviewer 5: Looks odd in the preview
<img width="372" alt="Screen Shot 2023-05-01 at 5 38 40 PM" src="https://user-images.githubusercontent.com/109224841/235536184-50d6b114-0716-4ecb-8511-791831eedf54.png">

| Description        | The maximum number of characters allowed is 300.                                                                                                  |

Follow this template to define the `media` object in the `manifest.json` file which includes an image, a video thumbnail, and a video:

{{< code-block lang="json" filename="manifest.json" collapsible="true" >}}
"media": [
      {
        "media_type": "image",
        "caption": "A Datadog Marketplace Integration OOTB Dashboard",
        "image_url": "images/integration_name_image_name.png"
      },
      {
        "media_type": "video",
        "caption": "A Datadog Marketplace Integration Overview Video",
        "image_url": "images/integration_name_video_thumbnail.png",
        "vimeo_id": 123456789
      },
    ],
{{< /code-block >}}

For more information, see [Integrations Assets Reference][22].

## Open a pull request

Before you open a pull request, run the following command to catch any problems with your integration:
```
ddev validate all <INTEGRATION_NAME>
```

> Next, commit all changes to your feature branch and push your changes to the remote repository. Open a pull request that contains your integration tile's asset files (including images) in the [`marketplace`][18] or [`integrations-extras`][26] repository. After you've created your pull request, automatic checks will run to verify that your pull request is in good shape and contains all the required content to be updated.

Reviewer 5: Can this be broken up into steps? It'll be easier for partners to parse.

---
title: Create a Tile
type: documentation
description: Learn how to develop and publish an integration tile.
aliases:
- 'developers/marketplace/offering'

> dependencies: "https://github.com/DataDog/integrations-core/blob/alai97/add-marketplace-documentation/docs/dev/create_a_tile.md"

Author: ```suggestion
dependencies: "https://github.com/DataDog/integrations-core/blob/master/docs/dev/create_a_tile.md"
```

---

> title: Create a Tile

Reviewer: Why does this page have the Datadog Sites dropdown? It doesn't affect anything does it?

Reviewer 3: @alai97

Author: They all have the Datadog Sites dropdown, it doesn't affect anything assuming that the content is the same throughout all data centers.

  tag: "Documentation"
  text: "Create an offering on the Datadog Marketplace"
---

## Overview

This page walks Technology Partners through how to create the tile that represents an offering that will appear on the **Integrations** or **Marketplace** page.

## Integration tiles

The tile serves as a point of entry where customers can learn about your offering, see setup instructions, and install or purchase your offering to unlock out-of-the-box dashboards and additional assets. 

{{< img src="developers/integrations/marketplace_or_integrations_tile.png" alt="An expanded tile modal of an example offering on the Integrations or Marketplace page" style="width:100%" >}}

* For any offerings that **do not use** the Datadog Agent—including API-based integrations, professional services listings, and software licenses—you will only need to create a tile and submit the tile-related files in order to publish your offering. This is called a _tile-only-listing_. Tile-ony listings apply in situations where Datadog does not host any of the code associated with the API-based integrations, and the other supported offering types do not require any code. 

> * For **Agent-based integrations**, however, you will need to create a tile, and additionally, submit all of your integration-related code (as well as your tile-related files) in one pull request. For more information, see [Create an Agent-based integration][27].

Reviewer: ```suggestion
* For **Agent-based integrations**, you will need to create a tile, and additionally, submit all of your integration-related code (as well as your tile-related files) in one pull request. For more information, see [Create an Agent-based integration][27].
```

* For any offerings that **do not use** the Datadog Agent—including API-based integrations, professional services listings, and software licenses—you will only need to create a tile and submit the tile-related files in order to publish your offering. This is called a _tile-only-listing_. Tile-ony listings apply in situations where Datadog does not host any of the code associated with the API-based integrations, and the other supported offering types do not require any code. 

* For **Agent-based integrations**, you will need to create a tile, and additionally, submit all of your integration-related code (as well as your tile-related files) in one pull request. For more information, see [Create an Agent-based integration][27].

<div class="alert alert-info">Select a tab for instructions on creating a tile on the Integrations or Marketplace page.</div>

{{< tabs >}}

{{% tab "Build a tile on the Integrations page" %}}

{{< img src="developers/integrations/integration_tile.png" alt="A tile representing an example offering on the Integrations page" style="width:25%" >}}

To build a tile on the [**Integrations** page][103]:

> <div class="alert alert-warning">If you have already gone through the steps to create an Agent integration and have built out a scaffolding, you can skip directly to [completing the necessary integration asset files](#complete-the-necessary-integration-asset-files). 

Author: Ahhh this one is hard, you have to hardcode the URL in!
```suggestion
<div class="alert alert-warning">If you have already gone through the steps to create an Agent integration and have built out a scaffolding, you can skip directly to <a href="https://docs.datadoghq.com/developers/integrations/create_a_tile/#complete-the-necessary-integration-asset-files">completing the necessary integration asset files</a>. 
```

If you used a directory other than `$HOME/dd` to clone the `integrations-extras` directory, use the following command to set your working repository:

{{< code-block lang="shell" >}}
ddev config set extras <PATH/TO/INTEGRATIONS_EXTRAS>
ddev config set repo extras
{{< /code-block >}}

## Populate the integration tile scaffolding

For Datadog API integrations that will be available out-of-the-box on the [Integrations page][102], use the Datadog Development Toolkit to create scaffolding for a tile-only listing.

1. Make sure you're inside the `integrations-extras` directory:

   ```shell
   cd $HOME/dd/integrations-extras
   ```

1. Run the `ddev` command with the `-t tile` option:

   ```shell
   ddev create -t tile "<Offering Name>"
   ```

[101]: https://docs.datadoghq.com/developers/integrations/python
[102]: https://github.com/Datadog/integrations-extras
[103]: https://app.datadoghq.com/integrations

{{% /tab %}}

{{% tab "Build a tile on the Marketplace" %}}

{{< img src="developers/integrations/marketplace_tile.png" alt="A tile representing an example offering on the Marketplace page" style="width:30%" >}}

To build a tile on the [**Marketplace** page][104]: 

> <div class="alert alert-warning">If you have already gone through the steps to create an Agent integration and have built out a scaffolding, you can skip directly to [completing the necessary integration asset files](#complete-the-necessary-integration-asset-files). 

Author: ```suggestion
<div class="alert alert-warning">If you have already gone through the steps to create an Agent integration and have built out a scaffolding, you can skip directly to <a href="https://docs.datadoghq.com/developers/integrations/create_a_tile/#complete-the-necessary-integration-asset-files">completing the necessary integration asset files</a>. 
```

> <div class="alert alert-warning">If you have already gone through the steps to create an Agent integration and have built out a scaffolding, you can skip directly to <a href="https://docs.datadoghq.com/developers/integrations/create_a_tile/#complete-the-necessary-integration-asset-files">completing the necessary integration asset files</a>. 

Reviewer 4: ```suggestion
<div class="alert alert-warning">If you have already gone through the steps to create an Agent integration and have built out the scaffolding, you can skip directly to <a href="#complete-the-necessary-integration-asset-files">completing the necessary integration asset files</a>. 
```

   
   ```shell
   git clone git@github.com:DataDog/marketplace.git
   ```

1. Create a feature branch to work in:

    ```shell
    git switch -c <YOUR INTEGRATION NAME> origin/master
    ```

## Install and configure the Datadog development toolkit

The Agent Integration Developer Tool allows you to create scaffolding when you are developing an integration by generating a skeleton of your integration tile's assets and metadata. For instructions on installing the tool, see [Install the Datadog Agent Integration Developer Tool][103].

Once you have installed the Agent Integration Developer Tool, configure it for the `marketplace` repository:

Set `marketplace` as the default working repository:

{{< code-block lang="shell" >}}
ddev config set marketplace $HOME/dd/marketplace
ddev config set repo marketplace
{{< /code-block >}}

If you used a directory other than `$HOME/dd` to clone the `marketplace` directory, use the following command to set your working repository:

{{< code-block lang="shell" >}}
ddev config set marketplace <PATH/TO/MARKETPLACE>
ddev config set repo marketplace

> {{< /code-block >}}

Reviewer 4: ~~~suggestion
```shell
ddev config set marketplace $HOME/dd/marketplace
ddev config set repo marketplace
```

If you used a directory other than `$HOME/dd` to clone the `marketplace` directory, use the following command to set your working repository:

```shell
ddev config set marketplace <PATH/TO/MARKETPLACE>
ddev config set repo marketplace
```
~~~

| Description        | The maximum number of characters allowed is 300.                                                                                                  |

Follow this template to define the `media` object in the `manifest.json` file which includes an image, a video thumbnail, and a video:

{{< code-block lang="json" filename="manifest.json" collapsible="true" >}}
"media": [
      {
        "media_type": "image",
        "caption": "A Datadog Marketplace Integration OOTB Dashboard",
        "image_url": "images/integration_name_image_name.png"
      },
      {
        "media_type": "video",
        "caption": "A Datadog Marketplace Integration Overview Video",
        "image_url": "images/integration_name_video_thumbnail.png",
        "vimeo_id": 123456789
      },
    ],
{{< /code-block >}}

For more information, see [Integrations Assets Reference][22].

## Open a pull request

Before you open a pull request, run the following command to catch any problems with your integration:

```
ddev validate all <INTEGRATION_NAME>
```

Complete the following steps:

1. Commit all changes to your feature branch.
2. Push your changes to the remote repository. 

> 3. Open a pull request that contains your integration tile's asset files (including images) in the [`marketplace`][18] or [`integrations-extras`][26] repository. 

Reviewer 4: If you don't want to mention both repos here, you could use tabs around the "Open a pull request" section

After you've created your pull request, automatic checks run to verify that your pull request is in good shape and contains all the required content to be updated.

## Review process

Once your pull request passes all checks, reviewers from the `Datadog/agent-integrations`, `Datadog/marketplace-review`, and `Datadog/documentation` teams provide suggestions and feedback on best practices.

> Once you have addressed the feedback and re-requested reviews, these reviewers approve your pull request. Contact the Marketplace team if you would like to preview the tile in your sandbox account. This allows you to validate and preview your tile before your tile is live to all customers.

Reviewer 4: ```suggestion
Once you have addressed the feedback and re-requested reviews, these reviewers approve your pull request. Contact the Marketplace team if you would like to preview the tile in your sandbox account. This allows you to validate and preview your tile before your tile goes live to all customers.
```

## Troubleshoot errors

Out-of-the-box integrations in the `integrations-extras` repository can run into validation errors when the forked repository is out of date with the origin. 

To resolve validation errors, update the forked repository on the GitHub web app: 

> 1. In [GitHub][29], navigate to your forked `integrations-extras` repository.
> 1. Click **Sync fork** under the green **Code** dropdown menu and click **Update branch**.

Reviewer 4: ```suggestion
1. In [GitHub][29], navigate to your forked `integrations-extras` repository.
1. Click **Sync fork** and click **Update branch**.
```
Rephrased cos this might make the user think the button is inside the drop-down menu. A screenshot might be helpful here.

To rebase and push changes:

1. Run `git checkout <your working branch>`.
1. Run `git rebase master`.

> 1. If there are any merge conflicts, resolve them. Then, run `git push origin <working branch> -f`.

Reviewer 4: ~~~suggestion
1. Update your local master branch:
   ```shell
   git checkout master
   git pull origin master
   ```
1. Merge `master` into your feature branch:
   ```shell
   git checkout <your working branch>
   git merge master
   ```
1. If there are any merge conflicts, resolve them. Then, run `git push origin <your working branch>`.
~~~
They should pull down the updated `master` branch before they rebase/merge, or nothing will happen. I don't know if I'd recommend a rebase over a standard master merge. If the user is not an experienced git user, a rebase could lead them into some trouble, especially if a lot of time has passed since they forked. It also avoids a history change and a force push.

- 'developers/marketplace/offering'
dependencies: "https://github.com/DataDog/integrations-core/blob/alai97/add-marketplace-documentation/docs/dev/create_a_tile.md"
further_reading:
- link: "https://www.datadoghq.com/blog/datadog-marketplace/"
  tag: "Blog"
  text: "Expand your monitoring reach with the Datadog Marketplace"
- link: "/developers/integrations/marketplace_offering"
  tag: "Documentation"
  text: "Create an offering on the Datadog Marketplace"
---

## Overview

This page walks Technology Partners through how to create the tile that represents an offering that will appear on the **Integrations** or **Marketplace** page.

## Integration tiles

The tile serves as a point of entry where customers can learn about your offering, see setup instructions, and install or purchase your offering to unlock out-of-the-box dashboards and additional assets. 

{{< img src="developers/marketplace/marketplace-tile-example.png" alt="Example Marketplace tile" style="width:30%" >}}

> * For any offerings that **do not use** the Datadog Agent—including API-based integrations, professional services listings, and software licenses—you will only need to create a tile and submit the tile-related files in order to publish your offering. This is called a _tile-only-listing_. Tile-ony listings apply in situations where Datadog does not host any of the code associated with the API-based integrations, and the supported offering types do not require any code. 

Reviewer 3: ```suggestion
* For any offerings that **do not use** the Datadog Agent—including API-based integrations, professional services listings, and software licenses—you will only need to create a tile and submit the tile-related files in order to publish your offering. This is called a _tile-only-listing_. Tile-ony listings apply in situations where Datadog does not host any of the code associated with the API-based integrations, and the other supported offering types do not require any code. 
```
